testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -1926,6 +1928,77 @@ public void testUpdateOnPartitionColumn()
         onSpark().executeQuery("DROP TABLE " + sparkTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testHandlingPartitionSchemaEvolutionInPartitionMetadata()
+    {
+        String baseTableName = "test_handling_partition_schema_evolution_" + randomTableSuffix();
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onTrino().executeQuery(format("CREATE TABLE %s (old_partition_key INT, new_partition_key INT, value date) WITH (PARTITIONING = array['old_partition_key'])", trinoTableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES (1, 10, date '2022-04-10'), (2, 20, date '2022-05-11'), (3, 30, date '2022-06-12'), (2, 20, date '2022-06-13')", trinoTableName));
+
+        onSpark().executeQuery(format("ALTER TABLE %s DROP PARTITION FIELD old_partition_key", sparkTableName));
+        onSpark().executeQuery(format("ALTER TABLE %s ADD PARTITION FIELD new_partition_key", sparkTableName));
+
+        List<Object> partitioning = onTrino().executeQuery(format("SELECT partition, record_count FROM iceberg.default.\"%s$partitions\"", baseTableName))
+                .column(1);
+        Set<String> partitions = partitioning.stream().map(String::valueOf).collect(toUnmodifiableSet());
+        Assertions.assertThat(partitions.size()).isEqualTo(3);
+        Assertions.assertThat(partitions).containsAll(ImmutableList.of(
+                "{old_partition_key=1, new_partition_key=null}",
+                "{old_partition_key=2, new_partition_key=null}",
+                "{old_partition_key=3, new_partition_key=null}"));
+
+        onSpark().executeQuery(format("ALTER TABLE %s DROP PARTITION FIELD new_partition_key", sparkTableName));
+        onSpark().executeQuery(format("ALTER TABLE %s ADD PARTITION FIELD old_partition_key", sparkTableName));
+
+        partitioning = onTrino().executeQuery(format("SELECT partition, record_count FROM iceberg.default.\"%s$partitions\"", baseTableName))
+                .column(1);
+        partitions = partitioning.stream().map(String::valueOf).collect(toUnmodifiableSet());
+        Assertions.assertThat(partitions.size()).isEqualTo(3);
+        Assertions.assertThat(partitions).containsAll(ImmutableList.of(
+                "{old_partition_key=1, new_partition_key=null}",
+                "{old_partition_key=2, new_partition_key=null}",
+                "{old_partition_key=3, new_partition_key=null}"));
+
+        onSpark().executeQuery(format("ALTER TABLE %s DROP PARTITION FIELD old_partition_key", sparkTableName));
```

Comment:
Can you add an insert along with each of these alter table sections so we can see the partition using this field show up?